src/main/java/com/powsybl/openloadflow/sensi/AcSensitivityAnalysis.java

```diff
@@ -147,26 +143,117 @@ private DenseMatrix initRhs(List<SensitivityFactor> factors, LfNetwork lfNetwork
+        );
+        // create AC engine
+        AcLoadFlowParameters acParameters = OpenLoadFlowProvider.createAcParameters(network, matrixFactory, lfParameters, lfParametersExt, true, true);
+        try (AcloadFlowEngine engine = new AcloadFlowEngine(lfNetwork, acParameters)) {
+
+            engine.run();
+
+            VariableSet variableSet = engine.getVariableSet();
+            EquationSystem equationSystem = engine.getEquationSystem();
+
+            List<LfSensitivityFactor<ClosedBranchSide1ActiveFlowEquationTerm>> lfFactors = factors.stream().map(factor -> LfSensitivityFactor.create(factor, network, lfNetwork, equationSystem, ClosedBranchSide1ActiveFlowEquationTerm.class)).collect(Collectors.toList());
+
+            // index factors by variable group to compute a minimal number of states
+            List<SensitivityFactorGroup> factorGroups = createFactorGroups(network, lfFactors);
+            if (factorGroups.isEmpty()) {
+                return Pair.of(Collections.emptyList(), Collections.emptyMap());
+            }
 
-        // filter invalid factors
-        List<SensitivityFactor> validFactors = validateFactors(network, factors, lfNetwork);
+            // compute the participation for each injection factor (+1 on the injection and then -participation factor on all
+            // buses that contain elements participating to slack distribution
+            List<ParticipatingElement> participatingElements = null;
+            Map<String, Double> slackParticipationByBus;
+            if (lfParameters.isDistributedSlack()) {
+                participatingElements = getParticipatingElements(lfNetwork, lfParameters, lfParametersExt);
+                slackParticipationByBus = participatingElements.stream().collect(Collectors.toMap(
+                    element -> element.getLfBus().getId(),
+                    element -> -element.getFactor(),
+                    Double::sum
+                ));
+            } else {
+                slackParticipationByBus = Collections.singletonMap(lfNetwork.getSlackBus().getId(), -1d);
+            }
+            computeInjectionFactors(slackParticipationByBus, factorGroups);
 
-        // create AC equation system
-        EquationSystem equationSystem = AcEquationSystem.create(lfNetwork, new VariableSet());
+            List<SensitivityValue> baseValues = new ArrayList<>();
 
-        // create jacobian matrix from current network state
-        VoltageInitializer voltageInitializer = getVoltageInitializer(lfParameters);
+            // we make the assumption that we ran a loadflow before, and thus this jacobian is the right one
+            try (JacobianMatrix j = createJacobianMatrix(equationSystem, new PreviousValueVoltageInitializer())) {
+                // otherwise, defining the rhs matrix will result in integer overflow
+                assert Integer.MAX_VALUE / (equationSystem.getSortedEquationsToSolve().size() * Double.BYTES) > factorGroups.size();
```

Comment:
@geofjamg it seems that we have an issue for matrix allocation, any idea to solve this?

Reply:
The solution is to solve by chunk, and chunk size should be a parameter (<= size(int). We can create a issue. for that too.